dev-client/src/screens/LocationScreens/LocationSoilIdScreen.tsx

```diff
@@ -45,29 +47,42 @@ export const LocationSoilIdScreen = ({siteId, coords}: Props) => {
     isSite ? selectSite(siteId)(state) : undefined,
   );
 
+  const siteIsRequiredButMissing = isSite && !site;
+  const handleMissingSite = useHandleMissingSiteOrProject();
+  const requirements = [
```

Comment:
Let's add a comment here / find some other way to make this logic easier to understand.